### PR TITLE
Add Shims for Spark 3.4.3

### LINKED ISSUE #10666: [FEA]  Create Spark 3.4.3 shim
**Is your feature request related to a problem? Please describe.**
Apache Spark is going to release a 3.4.3 with bug fixes. We need to add a shim if we want to support it.

### PR DESCRIPTION
**Pull request by razajafri**
This PR adds support for Spark 3.4.3. 

**Changes Made** 
* Generated shims using shimplify
* Modified the SparkShimServiceProvider to have the correct version 
* Modified the aggregator and parent poms 

**Tests** 
* Modified the SparkShimsSuite to test the correct version
* Ran all integration tests

fixes #10666

### CODE REVIEW

File: tools/generated_files/311/supportedExprs.csv
Comment (NVnavkumar) [inline tools/generated_files/311/supportedExprs.csv:1]: Any reason this file for 311 was updated?

Comment (razajafri) [reply tools/generated_files/311/supportedExprs.csv:1]: Good catch! Apparently, I had a 311 classes built from before this change got in so they weren't being replaced. Doing a clean and rebuilding everything undid this.

**Review by NvTimLiu — DISMISSED** [review]
LGTM from CI side.

File: pom.xml [hunk 1]
```diff
@@ -751,6 +771,7 @@
         <spark340.version>3.4.0</spark340.version>
         <spark341.version>3.4.1</spark341.version>
         <spark342.version>3.4.2</spark342.version>
+        <spark343.version>3.4.3-SNAPSHOT</spark343.version>
```
Comment (jlowe) [inline pom.xml:774]: 3.4.3 has been released.
```suggestion
        <spark343.version>3.4.3</spark343.version>
```

File: pom.xml [hunk 2]
```diff
@@ -810,6 +831,7 @@
             351
         </noSnapshot.buildvers>
         <snapshot.buildvers>
+            343
```
Comment (jlowe) [inline pom.xml:834]: This should be moved to the non-snapshot section

File: sql-plugin/src/main/spark343/scala/com/nvidia/spark/rapids/shims/spark343/SparkShimServiceProvider.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
```
Comment (jlowe) [inline sql-plugin/src/main/spark343/scala/com/nvidia/spark/rapids/shims/spark343/SparkShimServiceProvider.scala:2]:
```suggestion
 * Copyright (c) 2024, NVIDIA CORPORATION.
```

File: sql-plugin/src/main/spark343/scala/com/nvidia/spark/rapids/shims/spark343/SparkShimServiceProvider.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+/*** spark-rapids-shim-json-lines
+{"spark": "343"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.shims.spark343
+
+import com.nvidia.spark.rapids.SparkShimVersion
+
+object SparkShimServiceProvider {
+  val VERSION = SparkShimVersion(3, 4, 3)
+  val VERSIONNAMES = Seq(s"$VERSION", s"$VERSION-SNAPSHOT")
```
Comment (jlowe) [inline sql-plugin/src/main/spark343/scala/com/nvidia/spark/rapids/shims/spark343/SparkShimServiceProvider.scala:26]:
```suggestion
  val VERSIONNAMES = Seq(s"$VERSION")
```

File: sql-plugin/src/main/spark343/scala/com/nvidia/spark/rapids/spark343/RapidsShuffleManager.scala [hunk 5]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
```
Comment (jlowe) [inline sql-plugin/src/main/spark343/scala/com/nvidia/spark/rapids/spark343/RapidsShuffleManager.scala:2]:
```suggestion
 * Copyright (c) 2024, NVIDIA CORPORATION.
```

File: sql-plugin/src/test/spark343/scala/com/nvidia/spark/rapids/shims/spark343/SparkShimsSuite.scala [hunk 6]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
```
Comment (jlowe) [inline sql-plugin/src/test/spark343/scala/com/nvidia/spark/rapids/shims/spark343/SparkShimsSuite.scala:2]:
```suggestion
 * Copyright (c) 2024, NVIDIA CORPORATION.
```

File: tools/pom.xml [hunk 7]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (jlowe) [inline tools/pom.xml:1]:
```suggestion
```

**Comment by razajafri** [comment]
@NVnavkumar @jlowe Thanks for the review. I think I have addressed your concerns PTAL

**Review by jlowe — APPROVED** [review]

**Comment by NvTimLiu** [comment]
We'll wait to trigger pre-merge CI after nightly spark343 private shim get uploaded to OSS maven snapshot repo: https://oss.sonatype.org/content/repositories/snapshots/com/nvidia/rapids-4-spark-private-parent_2.12/24.06.0-SNAPSHOT/

**Comment by razajafri** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

**Pull request merged**